src/components/melding/NyMelding.tsx

```diff
@@ -60,12 +61,39 @@ function NyMelding({ lukkeKnapp }: NyMeldingProps) {
         },
         onSubmit: ({ value }) => {
             const body = generateRequestBody(value as NyMeldingSchema);
-            mutate({ body: body });
+            mutate(
+                { body: body },
+                {
+                    onSuccess: () => {
+                        removeDraft();
+                        form.reset(
+                            {
+                                ...defaultFormOptions,
+                                melding: '',
+                                meldingsType
+                            },
+                            { keepDefaultValues: true }
+                        );
+                    }
+                }
+            );
         }
     });
 
     const meldingsType = useStore(form.store, (state) => state.values.meldingsType);
     const meldingsTypeTekst = meldingsTyperTekst[meldingsType];
+    const melding = useStore(form.store, (state) => state.values.melding);
+    const meldingFieldMeta = useStore(form.store, (state) => state.fieldMeta.melding);
+
+    useEffect(() => {
+        // Hvis bruker fjerner alt innhold i meldingen, fjern draften
+        if (meldingFieldMeta?.isDirty && melding.length === 0) {
+            removeDraft();
+        }
+        if (melding.length > 0) {
+            updateDraft(melding);
+        }
+    }, [updateDraft, removeDraft, melding, meldingFieldMeta]);
```

Comment:
hmm, er ikke alt for fan av denne løsningen :thinking:

Mest pga useEffecten kjører ved hver endring i meldingen og trigger en re-render, og at den ikke er debouncet.

Er det mulig å legge inn dette i `onChange`? Med debounce? 

Ser at det står om side effects i docsa: https://tanstack.com/form/latest/docs/framework/react/guides/listeners (dette skal i det minste fjerne re-rendering issuet).

Reply:
De skal og ha innebygget debouncing for onChange listeneres og validation på field-nivå. Se eks: https://tanstack.com/form/latest/docs/reference/interfaces/formoptions#asyncdebouncems

Reply:
ah okey ja det funker :) 

Men ja enten put den inn i en `form.store.subscribe()` eller i en listener på feltet. Jeg tenker jo at det er cleanere med en listener på feltet og ikke ha subscription i en useEffect.